airbyte-webapp/.gitignore

```diff
@@ -24,8 +24,12 @@ yarn-error.log*
 /.idea
 
 .npmrc
+
+# Environment overwrites
 .env.development
 .env.production
+.experiments.dev
+.features.dev
```

Comment:
I don't feel super strongly about this, but: I would prefer keeping all environment variable overrides in one place, e.g. `.env.development`. My thinking is that the more gitignored environment override files we have, the more complex it is to know which values are coming from where, and the more potential for a confusing local dev experience.

Is there a reason why we don't just do something like `REACT_APP_FEATURE_FOO=bar` and `REACT_APP_EXPERIMENT_FOO=bar` in `.env.development`? The only blocker I see is that experiments are defined as a TS interface, so we would potentially need a different structure there so we can loop over it and look for overrides (instead of defining `REACT_APP_EXPERIMENT_OVERWRITES` as a JSON object).

Reply:
There are 2 main issues just using env files for this only:

* The main reason against that was, that ids of experiments are dot notated camel cased, which might lead to the situation that we might need to have properties either called `REACT_APP_EXPERIMENT_connector.SomeThing=true`, which looks just super weird, or we'd have some magic logic converting that into upper snake case, but then you'd need to implicitally understand how that would be looking alike, would it be `REACT_APP_EXPERIMENT_CONNECTOR_SOME_THING` or `REACT_APP_EXPERIMENT_CONNECTOR_SOMETHING`?
* Experiments (in contrast to features, who're just on/off) can support JSON values. We need some logic to handle that when overwriting. Saying we always do a `JSON.parse` on them, causes us to also need to quote plain string values, which might be non intuitive either e.g. `REACT_APP_EXPERIMENT_STRING_VALUE="foo"`, since all other string values in the env file won't need to be quoted.

I don't have strong opinions around keeping everything in one file together (as long as they following the same logic for all keys inside it), but know that this came up as a problem with the rest of the chapter, so maybe you could raise that question in the chapter slack to get more feedback on whether we want to go this route or not?

Reply:
Right, the JSON values make sense why we would want to have a separate file. From this perspective keeping them in one file does not seem like the best solution to me. I think I'd either be in favor of your original syntax in `.experiments.dev` or something like `.experiments.json`:

```json
{
  "authPage.signup.hideName": "true",
  "authPage.oauth.position": "top",
  "connector.orderOverwrite": { "connectorId": 5 }
}
```

Which is maybe a little easier to read/write because you get native JSON syntax highlighting in your IDE.